sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -221,6 +224,7 @@ class program_impl {
     if (!is_host()) {
       create_cl_program_with_source(KernelSource);
       build(BuildOptions);
+      AllowKernelsCaching = false;
```

Comment:
I believe this line should be at the beginning of the function. Currently caching happens  before this flag is set for false.

Reply:
This is true that this line should be at the beginning of the function.
Still there is no caching done here in create_cl_program_with_source or build methods.